src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/scheduling/GlobalConfChecker.java

```diff
@@ -0,0 +1,265 @@
+
+        ServerConfType serverConf = globalConfCheckerHelper.getServerConf();
+        SecurityServerId securityServerId = null;
+
+        try {
+            if (GlobalConfFacade.getServerOwner(buildSecurityServerId(serverConf)) == null) {
+                log.debug("Server owner not found in globalconf - owner may have changed");
+                updateOwner(serverConf);
+            }
+            securityServerId = buildSecurityServerId(serverConf);
+            log.debug("Security Server ID is \"{}\"", securityServerId);
+            updateClientStatuses(serverConf, securityServerId);
+            updateAuthCertStatuses(securityServerId);
+        } catch (Exception e) {
+            throw translateException(e);
+        }
+    }
+
+    private SecurityServerId buildSecurityServerId(ClientId ownerId, String serverCode) {
+        return SecurityServerId.create(
+                ownerId.getXRoadInstance(), ownerId.getMemberClass(),
+                ownerId.getMemberCode(), serverCode);
+    }
+
+    private SecurityServerId buildSecurityServerId(ServerConfType serverConf) {
+        ClientId ownerId = serverConf.getOwner().getIdentifier();
+        return buildSecurityServerId(ownerId, serverConf.getServerCode());
+    }
+
+    private void updateOwner(ServerConfType serverConf) throws Exception {
+        ClientId ownerId = serverConf.getOwner().getIdentifier();
+        for (ClientType client : serverConf.getClient()) {
+            // Look for another member that is not the owner
+            if (client.getIdentifier().getSubsystemCode() == null
+                    && !client.getIdentifier().equals(ownerId)) {
+                log.trace("Found potential new owner: \"{}\"", client.getIdentifier());
+
+                // Build a new server id using the alternative member as owner
+                SecurityServerId altSecurityServerId = buildSecurityServerId(client.getIdentifier(),
+                        serverConf.getServerCode());
+
+                // Get local auth cert
+                X509Certificate cert = getAuthCert(altSecurityServerId);
+
+                // Does the alternative server id exist in global conf?
+                // And does the local auth cert match with the auth cert of
+                // the alternative server from global conf?
+                if (GlobalConfFacade.getServerOwner(altSecurityServerId) != null
+                        && cert != null
+                        && altSecurityServerId.equals(globalConfFacade.getServerId(cert))
+                ) {
+                    log.debug("Set \"{}\" as new owner", client.getIdentifier());
+                    serverConf.setOwner(client);
+                }
+            }
+        }
+    }
+
+    private X509Certificate getAuthCert(SecurityServerId serverId) throws Exception {
+        log.debug("Get auth cert for security server '{}'", serverId);
+
+        AuthKeyInfo keyInfo = SignerClient.execute(new GetAuthKey(serverId));
```

Comment:
Wrapping the `SignerClient.execute` in the `SignerProxyFacade` would make this more testable.

Reply:
Fixed.